src/main/java/dev/vasyl/proj/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+package dev.vasyl.proj.service.impl;
+
+import dev.vasyl.proj.dto.shopping.cart.CartItemOperation;
+import dev.vasyl.proj.dto.shopping.cart.CartResponseDto;
+import dev.vasyl.proj.dto.shopping.cart.CreateCartItemRequestDto;
+import dev.vasyl.proj.dto.shopping.cart.UpdateCartItemRequestDto;
+import dev.vasyl.proj.exception.EntityNotFoundException;
+import dev.vasyl.proj.exception.InsufficientItemsException;
+import dev.vasyl.proj.mapper.ShoppingCartMapper;
+import dev.vasyl.proj.model.Book;
+import dev.vasyl.proj.model.CartItem;
+import dev.vasyl.proj.model.ShoppingCart;
+import dev.vasyl.proj.model.User;
+import dev.vasyl.proj.repository.BookRepository;
+import dev.vasyl.proj.repository.CartItemRepository;
+import dev.vasyl.proj.repository.ShoppingCartRepository;
+import dev.vasyl.proj.service.ShoppingCartService;
+import java.util.Optional;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private static final int DEFAULT_QUANTITY = 1;
+
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final BookRepository bookRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    public CartResponseDto getCart(User user) {
+        return shoppingCartMapper.toCartDto(getShoppingCartByUserId(user.getId()));
+    }
+
+    @Transactional
```

Comment:
You can move `@Transactional` above the class
Maybe this article will help you better understand how these annotations and pattern Proxy work:
https://medium.com/@denis.verkhovsky/spring-how-annotations-work-part-2-f9ea764abc28